### PR TITLE
Add fixed-point solver statistics

### PR DESCRIPTION
**Pull request by devmotion**
This PR supersedes https://github.com/JuliaDiffEq/DelayDiffEq.jl/pull/153, since most commits there are already merged into master.

### CODE REVIEW

File: src/ddestats.jl [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+  naccept::Int
+  nreject::Int
+  maxeig::Float64
+end
+
+DDEStats(x::Int = -1) = DDEStats(x, x, x, x, x, x, x, x, x, x, x, x, 0.0)
+
+function Base.show(io::IO, s::DDEStats)
+  println(io, summary(s))
+  @printf io "%-60s %-d\n" "Number of function 1 evaluations:" s.nf
```
Comment (devmotion) [inline src/ddestats.jl]: Should the numbers be left or right aligned?

**Comment by devmotion** [comment]
The Appveyor tests fail due to the known issues on Win32...

File: src/ddestats.jl (hunk 1 above)
Comment (ChrisRackauckas) [reply src/ddestats.jl]: I don't have an opinion on that.

File: src/ddestats.jl [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+  nsolve::Int
+  njacs::Int
+  nnonliniter::Int
+  nnonlinconvfail::Int
+  nfpiter::Int
+  nfpconvfail::Int
+  ncondition::Int
+  naccept::Int
+  nreject::Int
+  maxeig::Float64
```
Comment (ChrisRackauckas) [inline src/ddestats.jl]: We should double check whether this works with AD. IIRC there is an `internalnorm` before the save here so we're good.

Comment (devmotion) [reply src/ddestats.jl]: It's copied from DiffEqBase: https://github.com/JuliaDiffEq/DiffEqBase.jl/blob/master/src/destats.jl

Comment (ChrisRackauckas) [reply src/ddestats.jl]: okay awesome then it must work.

File: src/ddestats.jl (hunk 1 above)
Comment (devmotion) [reply src/ddestats.jl]: Should DiffEqBase use `@printf` as well so the numbers are aligned?

Comment (ChrisRackauckas) [reply src/ddestats.jl]: Yeah it probably should. That would make it much nicer.

**Pull request merged**